File: math/1344_angle_between_hands_of_a_clock/1344_angle_between_hands_of_a_clock.py
class Solution(object):
    def angleClock(self, hour, minutes):
        """
        :type hour: int
        :type minutes: int
        :rtype: float
        """
        m = minutes * (360/60)
        if hour == 12:
            h = minutes * (360.0/12/60)
        else:
            h = hour * (360/12) + minutes * (360.0/12/60)
        res = abs(m-h)
        return 360 - res if res > 180 else res


hour = 12
minutes = 30
res = Solution().angleClock(hour, minutes)
print(res)

File: math/1344_angle_between_hands_of_a_clock/1344_angle_between_hands_of_a_clock.go
package main

import "fmt"

func angleClock(hour int, minutes int) float64 {
    var (
        locaH, locaM, diff float64
    )

    locaM = float64(6 * minutes)
    locaH = float64(30*hour%360) + float64(minutes)*30.0/60.0

    diff = max(locaH, locaM) - min(locaH, locaM)
    if diff > 180.0 {
        diff = 360 - diff
    }
    return diff
}

func max(a, b float64) float64 {
    if a > b {
        return a
    }
    return b
}
func min(a, b float64) float64 {
    if a < b {
        return a
    }
    return b
}

func main() {
	hour := 12
	minutes := 30
	res := angleClock(hour, minutes)
	fmt.Println(res)
}
